services/src/main/java/org/keycloak/services/clientpolicy/executor/SecureRequestObjectExecutor.java

```diff
@@ -229,6 +243,12 @@ private void executeOnAuthorizationRequest(
             throw new ClientPolicyException(OAuthErrorException.INVALID_REQUEST, "Invalid parameter. Parameters in 'request' object not matching with request parameters");
         }
 
+        Boolean encryptionRequired = Optional.ofNullable(configuration.isEncryptionRequired()).orElse(Boolean.FALSE);
+        if (encryptionRequired && session.getAttribute(AuthzEndpointRequestParser.AUTHZ_REQUEST_OBJECT_ENCRYPTED) == null) {
```

Comment:
@pedroigor It seems to me that with this flag `encryptionRequired` set to true, all clients in the realm (which will pass the client policy conditions) will be required to use encrypted request objects. Which I am not sure if it is desired behaviour.

IMO it will be good to stick with the same like for request object signatures. Which means that it would be enforced per the client if request object is encrypted just in case that ` request.object.encryption.alg` is used for particular client. And this is already done in this PR (in AuthzEndpointRequestObjectParser).

In shortcut, my vote is to remove flag `encryptionRequired` from SecureRequestObjectExecutor. WDYT?

Reply:
As you mentioned, clients that match the policy conditions.

I've opened a discussion on keycloak-dev about this topic in order to understand whether a setting should be at the client or at the client policy.

I agree with @tnorimat that moving configuration to policies is the best to do. And whether encryption is required looks like a good example.

As per signature, I guess it was done before client policies support. Right?

I'm not sure, for me either way works. Let me know if the above makes sense if not I can change to do the same for signature.

Reply:
Yes, we discussed this already some time ago during work on client policies and during designing client policies. And I also agree that it will be nice to move many of client switches and keep them at the client policy level. This will allow some flexibility (EG. enforce secure ID Token signature algorithm just in case that `scope=fapi` is requested, which is possible due the fact that client policy condition will be evaluated just with the  `scope=fapi` and the particular executor for enforce algorithm will be also called just at that time etc).

However one thing is, that some configurations are enforced by OIDC specification itself at the client level (like the `request_object_encryption_alg` and `request_object_encryption_enc`) and hence for those switches we need to stick them at the client level exactly as you did in this PR.

I am personally still not sure if the particular switch `encryptionRequired` is needed on the executor as we don't have similar thing for the signature, but I don't see this as a blocker in this PR as it is false by default anyway. so I am approving this. Thanks for the updates!